packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -397,12 +397,21 @@ export default class CodeGen {
         );
         const listenerObjectAST = objectToAST(listenerObj, (key) => {
             const componentHandler = this.bindExpression(listenerObj[key].handler);
-            const handler = this.genBind(componentHandler);
-
-            return memorizeHandler(this, componentHandler, handler);
+            return this.genBind(componentHandler);
         });
 
-        return t.property(t.identifier('on'), listenerObjectAST);
+        // Generate a unique identifier for the `on` object
+        const onObjectId = this.getMemorizationId();
+
+        // Cache the `on` object on the `$ctx` object
+        return t.property(
+            t.identifier('on'),
+            t.assignmentExpression(
+                '||=',
+                t.memberExpression(t.identifier(TEMPLATE_PARAMS.CONTEXT), onObjectId),
+                listenerObjectAST
+            )
```

Comment:
Or assignment generates less code. It's available since chrome 85.

Is this fine? Or should I stick with short circuit (chrome 1)?